src/server/svr_jobfunc.c

```diff
@@ -278,11 +274,7 @@ svr_enquejob(job *pjob)
 
 			if (is_linked(&svr_alljobs, &pjob->ji_alljobs) == 0) {
 				append_link(&svr_alljobs, &pjob->ji_alljobs, pjob);
-				/**
-				 * Add to AVL tree so that find_job() can return
-				 * faster compared to linked list traverse.
-				 */
-				svr_avljob_oper(pjob, 0);
+				tree_add_del(AVL_jobs, (void *)pjob->ji_qs.ji_jobid, (void *)pjob, TREE_OP_ADD);
```

Comment:
is there any reason that when we add/remove a job in the linklist, we don't have to do it in the tree as well?
I think you can just make a function which will add/remove the job from both places at once and use that function.
This way nobody has to worry about missing on updating one of them
Same comment will apply to queue/reservations as well.

Reply:
> is there any reason that when we add/remove a job in the linklist, we don't have to do it in the tree as well?

Yes, some times job is linked in another linklist like svr_newjobs (or maybe some other linklist) and we have to add job in AVL only when we link/unlink from svr_alljobs not when we link/unlink job from other linklist.



> I think you can just make a function which will add/remove the job from both places at once and use that function.
> This way nobody has to worry about missing on updating one of them
> Same comment will apply to queue/reservations as well.

We can't right, because of the above reason.

Reply:
Like we discussed offline, job in svr_newjobs list is very short lived. It will be better if we add a function that unlink the job from the list as well as from the tree. Four cases where job is in svr_newjobs list it will anyways not be part of the tree so it’s ok to unlink it directly from that list.